### ISSUE TITLE
ASSERT Character Selection Bug and Exit error

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#134]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/134) by Raster Ron (Bitbucket: [rasteron](https://bitbucket.org/rasteron), GitHub: [rasteron](https://github.com/rasteron)).**

----------------------------------------

Hi, 

I'm having problems logging back to character selection screen or if you quit there's this error I'm getting in Windows client. I noticed this started to happen since the latest patch or update. Other players seem to experience this problem as well.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Next time, right click, select all, right click, copy. And paste here entirely. Can be handy to have the whole text. Thanks!

**Comment by ryzom-pipeline** [comment]
**Original comment by Raster Ron (Bitbucket: [rasteron](https://bitbucket.org/rasteron), GitHub: [rasteron](https://github.com/rasteron)).**

----------------------------------------

Oh ok, i forgot. here it is:

```
A failed assertion occurs
ProcName: client_ryzom_rd.exe
Date: 2014/02/17 10:32:12
File: Z:\home\nevrax\code\ryzom\client\src\interface_v3\view_renderer.cpp
Line: 967
FuncName: CViewRenderer::deleteTexture
Reason: "(uint)textureId < _SImageIterators.size()" 
-------------------------------

Log with no filter:
-------------------------------
2014/02/17 10:32:05 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 2), 1 actions (ReceivedAck=271/lastAck=261/nextSentPacket=273)
2014/02/17 10:32:05 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:05 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 1)
2014/02/17 10:32:05 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 1), 1 actions (ReceivedAck=274/lastAck=264/nextSentPacket=276)
2014/02/17 10:32:05 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:06 client_ryzom_rd.exe DBG   498 session_browser.cpp 145 : SB: Clearing SendQueue
2014/02/17 10:32:06 client_ryzom_rd.exe DBG   498 session_browser.cpp 183 : SB: Connecting...
2014/02/17 10:32:06 client_ryzom_rd.exe DBG  1124 session_browser_impl.cpp 250 : SB: on_connectionFailed
2014/02/17 10:32:06 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 0)
2014/02/17 10:32:06 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 0), 1 actions (ReceivedAck=277/lastAck=267/nextSentPacket=279)
2014/02/17 10:32:06 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:07 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 3)
2014/02/17 10:32:07 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 3), 1 actions (ReceivedAck=280/lastAck=270/nextSentPacket=282)
2014/02/17 10:32:07 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:07 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 2)
2014/02/17 10:32:07 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 2), 1 actions (ReceivedAck=283/lastAck=273/nextSentPacket=285)
2014/02/17 10:32:07 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:08 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 1)
2014/02/17 10:32:08 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 1), 1 actions (ReceivedAck=286/lastAck=276/nextSentPacket=288)
2014/02/17 10:32:08 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:08 client_ryzom_rd.exe DBG   498 session_browser.cpp 145 : SB: Clearing SendQueue
2014/02/17 10:32:08 client_ryzom_rd.exe DBG   498 session_browser.cpp 183 : SB: Connecting...
2014/02/17 10:32:08 client_ryzom_rd.exe DBG  1124 session_browser_impl.cpp 250 : SB: on_connectionFailed
2014/02/17 10:32:08 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 0)
2014/02/17 10:32:08 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 0), 1 actions (ReceivedAck=289/lastAck=279/nextSentPacket=291)
2014/02/17 10:32:08 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:09 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 3)
2014/02/17 10:32:09 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 3), 1 actions (ReceivedAck=292/lastAck=282/nextSentPacket=294)
2014/02/17 10:32:09 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:10 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 2)
2014/02/17 10:32:10 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 2), 1 actions (ReceivedAck=295/lastAck=285/nextSentPacket=297)
2014/02/17 10:32:10 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:10 client_ryzom_rd.exe DBG   498 session_browser.cpp 145 : SB: Clearing SendQueue
2014/02/17 10:32:10 client_ryzom_rd.exe DBG   498 session_browser.cpp 183 : SB: Connecting...
2014/02/17 10:32:10 client_ryzom_rd.exe DBG  1124 session_browser_impl.cpp 250 : SB: on_connectionFailed
2014/02/17 10:32:10 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 1)
2014/02/17 10:32:10 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 1), 1 actions (ReceivedAck=298/lastAck=288/nextSentPacket=300)
2014/02/17 10:32:10 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:11 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 0)
2014/02/17 10:32:11 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 0), 1 actions (ReceivedAck=301/lastAck=291/nextSentPacket=303)
2014/02/17 10:32:11 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:11 client_ryzom_rd.exe INF  1124 action_handler_game.cpp 1211 : User Force to Quit ryzom
2014/02/17 10:32:11 client_ryzom_rd.exe INF  1124 impulse_decoder.cpp 82 : CLIMPD: received new impulsion 1 (len=131) at level 2 (channel 3)
2014/02/17 10:32:11 client_ryzom_rd.exe DBG  1124 impulse_decoder.cpp 93 : CLIMPD: at level 2 (channel 3), 1 actions (ReceivedAck=304/lastAck=294/nextSentPacket=306)
2014/02/17 10:32:11 client_ryzom_rd.exe DBG  1124 cdb_synchronised.cpp 180 : Update DB
2014/02/17 10:32:11 client_ryzom_rd.exe INF  1124 client.cpp 602 : Main loop of Ryzom success
2014/02/17 10:32:11 client_ryzom_rd.exe INF  1124 client.cpp 634 : Releasing of Ryzom...
2014/02/17 10:32:11 client_ryzom_rd.exe WRN  1124 editor.cpp 3928 : *R2* release
2014/02/17 10:32:11 client_ryzom_rd.exe WRN  1124 editor.cpp 3928 : *R2* release
2014/02/17 10:32:11 client_ryzom_rd.exe INF  1124 network_connection.cpp 2764 : CNET[022288F8]: sent DISCONNECTION
2014/02/17 10:32:12 client_ryzom_rd.exe AST  1124 view_renderer.cpp 967 : "(uint)textureId < _SImageIterators.size()" 
```

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

And what steps to reproduce this error?

**Comment by ryzom-pipeline** [comment]
**Original comment by Raster Ron (Bitbucket: [rasteron](https://bitbucket.org/rasteron), GitHub: [rasteron](https://github.com/rasteron)).**

----------------------------------------

If I logged in and press quit to Character Selection, it doesn't work. If I continue to quit I'm always getting this error.

**Comment by ryzom-pipeline** [comment]
**Original comment by Łukasz K. (Bitbucket: [maczuga](https://bitbucket.org/maczuga), GitHub: [maczuga](https://github.com/maczuga)).**

----------------------------------------

I can confirm this, same error.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Does this occur only on WG's shard or on any shard?

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Returning to character selection does not work on WG shard, likely due to SBS misconfiguration.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Different error here before your assert occurs.

```
#!lua

A failed assertion occurs
ProcName: ryzom_client_r.exe
Date: 2014/02/17 13:36:37
File: R:\code\ryzom\client\src\interface_v3\flying_text_manager.cpp
Line: 44
FuncName: CFlyingTextManager::~CFlyingTextManager
Reason: "_InScenePool.empty() && _InSceneCurrent.empty()" 
-------------------------------

Log with no filter:
-------------------------------
2014/02/17 13:36:35 ryzom_client_r.exe INF   54c action_handler_game.cpp 1214 : User Force to Quit ryzom
2014/02/17 13:36:35 ryzom_client_r.exe INF   54c client.cpp 617 : Main loop of Ryzom success
2014/02/17 13:36:35 ryzom_client_r.exe INF   54c client.cpp 648 : Releasing of Ryzom...
2014/02/17 13:36:36 ryzom_client_r.exe WRN   54c editor.cpp 3924 : *R2* release
2014/02/17 13:36:36 ryzom_client_r.exe WRN   54c editor.cpp 3924 : *R2* release
2014/02/17 13:36:36 ryzom_client_r.exe INF   54c network_connection.cpp 2762 : CNET[01710288]: sent DISCONNECTION
2014/02/17 13:36:36 ryzom_client_r.exe INF   54c sound_driver_xaudio2.cpp 196 : XA2: Releasing CSoundDriverXAudio2
2014/02/17 13:36:36 ryzom_client_r.exe WRN   54c sound_driver_xaudio2.cpp 226 : XA2: _Listener: !NULL
2014/02/17 13:36:36 ryzom_client_r.exe INF   54c sound_driver_xaudio2.cpp 186 : XA2: Destroying CSoundDriverXAudio2
2014/02/17 13:36:36 ryzom_client_r.exe WRN   54c listener_xaudio2.cpp 72 : XA2: Destroying CListenerXAudio2
2014/02/17 13:36:36 ryzom_client_r.exe DBG   54c release.cpp 572 : VR [C]: VR Shutting down
2014/02/17 13:36:36 ryzom_client_r.exe DBG   e70 session_browser.cpp 145 : SB: Clearing SendQueue
2014/02/17 13:36:36 ryzom_client_r.exe DBG   e70 session_browser.cpp 183 : SB: Connecting...
2014/02/17 13:36:37 ryzom_client_r.exe AST   54c flying_text_manager.cpp 44 : "_InScenePool.empty() && _InSceneCurrent.empty()" 
-------------------------------
User Crash Callback:
-------------------------------
ConnectState: Disconnect
NumServerHOP: 0
NumFarTP: 0
NumReselectPerso: 0
-------------------------------

```

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Fix #134 Exit crash after failed response from character selection

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Character selection on WG's shard seems to be functioning again as well, after the shard reboot that just happened.

**Comment by ryzom-pipeline** [comment]
**Original comment by Raster Ron (Bitbucket: [rasteron](https://bitbucket.org/rasteron), GitHub: [rasteron](https://github.com/rasteron)).**

----------------------------------------

This works great now, tested earlier both quit and character selection. Thanks Jan.

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

Fix #134 Exit crash after failed response from character selection
